x-pack/plugins/lens/public/datasources/text_based/to_expression.ts

```diff
@@ -24,23 +25,65 @@ function getExpressionForLayer(
   layer.columns.forEach((col) => {
     if (idMapper[col.fieldName]) {
       idMapper[col.fieldName].push({
+        // ...col,
         id: col.columnId,
-        label: col.fieldName,
+        label: col.customLabel ? col.label : col.fieldName,
       } as OriginalColumn);
     } else {
       idMapper = {
         ...idMapper,
         [col.fieldName]: [
           {
+            // ...col,
             id: col.columnId,
-            label: col.fieldName,
+            label: col.customLabel ? col.label : col.fieldName,
           } as OriginalColumn,
         ],
       };
     }
   });
   const timeFieldName = layer.timeField ?? undefined;
 
+  const formatterOverrides: ExpressionAstFunction[] = layer.columns
+    .filter((col) => col.params?.format)
+    .map((col) => {
+      // TODO: improve the type handling here
+      const format = col.params!.format!;
```

Comment:
```suggestion
    .filter((col): col is TextBasedLayerColumn & {params: {format: ValueFormatConfig}}  => col.params?.format)
    .map((col) => {
      const format = col.params.format;
```

Reply:
this doesnt seem to work
Type 'ValueFormatConfig | undefined' is not assignable to type 'boolean'.
Type 'undefined' is not assignable to type 'boolean'.

Reply:
```
 .filter((col): col is TextBasedLayerColumn & { params: { format: ValueFormatConfig } } =>
      Boolean(col.params?.format)
    )
```

Reply:
is this really better from what we currently have ?

Reply:
This creates a type guard that is the way to fix `col.params!.format!;` in this place.
You can move the filter/typeguard function outside this code to make it a bit less verbose.